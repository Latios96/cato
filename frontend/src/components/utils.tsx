import React from "react";
export function renderIf<T>(
  value: T,
  renderCallback: (value: T) => JSX.Element
): JSX.Element {
  if (value) {
    return renderCallback(value);
  }
  return <React.Fragment />;
}
