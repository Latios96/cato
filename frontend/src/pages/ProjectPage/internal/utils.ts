export function toCommaSeparatedString(strings: Set<string>) {
  return Array.from(strings).sort().join(",");
}

export function fromCommaSeparatedString(theString: string) {
  return new Set(theString.split(","));
}
