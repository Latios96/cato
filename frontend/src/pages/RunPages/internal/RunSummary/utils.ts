import { RunAggregate } from "../../../../catoapimodels/catoapimodels";

export interface StatusPercentage {
  waitingToStart: number;
  running: number;
  succeeded: number;
  failed: number;
}
export function calculateStatusPercentage(
  runAggregate: Pick<RunAggregate, "testCount" | "progress">
): StatusPercentage {
  return {
    waitingToStart:
      (runAggregate.progress.waitingTestCount / runAggregate.testCount) * 100,
    running:
      (runAggregate.progress.runningTestCount / runAggregate.testCount) * 100,
    succeeded:
      (runAggregate.progress.succeededTestCount / runAggregate.testCount) * 100,
    failed:
      (runAggregate.progress.failedTestCount / runAggregate.testCount) * 100,
  };
}
