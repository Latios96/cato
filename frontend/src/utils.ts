import ago from "s-ago";
import humanizeDuration from "humanize-duration";
import queryString from "query-string";

export function formatTime(datestr: string | null | undefined): string {
  if (!datestr) {
    return "";
  }
  datestr = datestr.replace(" GMT", "");
  var date = new Date(datestr);
  return ago(date);
}

export function formatDuration(
  durationInSeconds: number | "NaN" | undefined | null
): string {
  if (durationInSeconds == null || durationInSeconds === "NaN") {
    return "";
  }
  return humanizeDuration(durationInSeconds * 1000, { round: true });
}

export function updateQueryString(theQueryString: string, data: object) {
  const parsedData = queryString.parse(theQueryString);
  Object.assign(parsedData, data);
  return queryString.stringify(parsedData);
}
export function popFromQueryString(theQueryString: string, keys: string[]) {
  const parsedData = queryString.parse(theQueryString);
  keys.forEach((key) => {
    delete parsedData[key];
  });
  return queryString.stringify(parsedData);
}
