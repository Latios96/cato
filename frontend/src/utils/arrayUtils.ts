export function isLast(index: number, arr: any[]) {
  return index + 1 === arr.length;
}

export function isNotLast(index: number, arr: any[]) {
  return !isLast(index, arr);
}
