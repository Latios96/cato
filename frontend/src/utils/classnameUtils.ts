export function joinClassnames(names: (string | any)[]) {
  return names
    .filter((s) => s)
    .filter((s) => s.constructor === String)
    .join(" ");
}
