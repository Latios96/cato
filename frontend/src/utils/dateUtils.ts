import ago from "s-ago";
import humanizeDuration from "humanize-duration";

export function formatTime(datestr: string | null | undefined): string {
  if (!datestr) {
    return "";
  }
  datestr = datestr.replace(" GMT", "");
  var date = new Date(datestr);
  return ago(date);
}

export function formatDuration(
  durationInSeconds: number | "NaN" | undefined | null
): string {
  if (durationInSeconds == null || durationInSeconds === "NaN") {
    return "";
  }
  if (durationInSeconds > 0 && durationInSeconds < 1) {
    return "less than one second";
  }
  return humanizeDuration(durationInSeconds * 1000, { round: true });
}
