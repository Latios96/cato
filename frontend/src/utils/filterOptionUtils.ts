import queryString from "query-string";
import { FilterOptions, StatusFilter } from "../models/FilterOptions";
import { TestFailureReasonDto } from "../catoapimodels";

export function filterOptionsFromQueryString(
  theQueryString: string
): FilterOptions {
  const parsedData = queryString.parse(theQueryString);
  let filterOptions = new FilterOptions();
  if (parsedData.statusFilter && !Array.isArray(parsedData.statusFilter)) {
    filterOptions = filterOptions.withChangedStatusFilter(
      StatusFilter[parsedData.statusFilter as keyof typeof StatusFilter]
    );
  }
  if (
    parsedData.failureReasonFilter &&
    !Array.isArray(parsedData.statusFilter) &&
    filterOptions.status === StatusFilter.FAILED
  ) {
    filterOptions = filterOptions.withChangedFailureReason(
      TestFailureReasonDto[
        parsedData.failureReasonFilter as keyof typeof TestFailureReasonDto
      ]
    );
  }
  return filterOptions;
}

export function filterOptionsForQueryString(filterOptions: FilterOptions) {
  return {
    statusFilter: filterOptions.status,
  };
}

export function filterOptionsToQueryString(filterOptions: FilterOptions) {
  return queryString.stringify(filterOptionsForQueryString(filterOptions));
}
