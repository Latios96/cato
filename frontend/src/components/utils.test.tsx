import { mount } from "enzyme";
import { renderIf } from "./utils";
import React from "react";

interface Props {
  value?: string;
}

const TestingComponent = (props: Props) => {
  return (
    <div>
      {renderIf(props.value, (v) => {
        return <h1>{v}</h1>;
      })}
    </div>
  );
};

describe("renderIf", () => {
  it("should render empty when value is falsy", () => {
    const testingComponent = mount(<TestingComponent />);
    expect(testingComponent).toMatchSnapshot();
  });

  it("should render callback result", () => {
    const testingComponent = mount(<TestingComponent value={"test"} />);
    expect(testingComponent).toMatchSnapshot();
  });
});
