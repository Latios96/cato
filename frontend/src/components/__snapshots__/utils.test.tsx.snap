// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderIf should render callback result 1`] = `
<div>
  <h1>
    test
  </h1>
</div>
`;

exports[`renderIf should render empty when value is falsy 1`] = `<div />`;
